web-api/src/main/java/com/graphhopper/util/CustomModel.java

```diff
@@ -112,17 +111,20 @@ public Map<String, JsonFeature> getAreas() {
         return areas;
     }
 
-    public CustomModel setDistanceInfluence(double distanceFactor) {
-        this.distanceInfluence = distanceFactor;
-        return this;
-    }
-
     public boolean hasDistanceInfluence() {
         return distanceInfluence != null;
     }
 
-    public double getDistanceInfluence() {
-        return hasDistanceInfluence() ? distanceInfluence : DEFAULT_DISTANCE_INFLUENCE;
+    public CustomModel setDistanceInfluence(Double distanceFactor) {
+        this.distanceInfluence = distanceFactor;
+        return this;
+    }
+
+    /**
+     * Check with #hasDistanceInfluence() to avoid null values
+     */
+    public Double getDistanceInfluence() {
```

Comment:
...or simply check for `getDistanceInfluence() == null`. Why shall we recommend to use `hasDistanceInfluence`?

Reply:
Then we can also just remove `hasDistanceInfluence`?

Reply:
You mean the pure existence of a `hasXYZ` method (assuming the programmer sees it) serves as a hint that the value could be null? Probably a matter of preference. For me this is just a distraction. When I see something like `hasDistanceInfluence` I have no idea what it is and need to open the method to see what it does. The most standard way to make it obvious something might be missing is probably `Optional<Double>`.

Reply:
We should either use `Optional` or `@Nullable` if we want to improve developer experience. I agree with @easbar that the method only adds noise.